osf/models/preprint_provider.py

```diff
@@ -111,6 +113,20 @@ def absolute_api_v2_url(self):
         path = '/preprint_providers/{}/'.format(self._id)
         return api_v2_url(path)
 
+    def save(self):
+        dirty_fields = self.get_dirty_fields()
+        old_id = dirty_fields['_id']
+        if old_id:
+            for permission_type in GROUPS.keys():
+                group_name = GROUP_FORMAT.format(provider_id=old_id, group=permission_type)
+                try:
+                    permission_group = Group.objects.get(name=group_name)
+                    permission_group.name = GROUP_FORMAT.format(provider_id=self._id, group=permission_type)
+                    permission_group.save()
+                except Group.DoesNotExist:
+                    pass
+
+        return super(PreprintProvider, self).save()
```

Comment:
Possibly related to the current travis failures, use the passed in add. args so you do not end up using defaults for the call. `super(PreprintProvider, self).save(*args, **kwargs)`